src/kernels/tensor_view.hpp

```diff
@@ -77,4 +77,20 @@ struct tensor_layout_t
     uint64_t layout[N];
 };
 
+template <typename T>
+__host__ __device__ T inline get2DVal(const T* x, tensor_view_t<2> x_tv, uint64_t idx)
```

Comment:
Please add N to the template to create a more general getval function.

Reply:
I have updated template to receive N